vhdl_syntax/src/parser/productions/names.rs

```diff
@@ -32,4 +74,480 @@ impl<T: TokenStream> Parser<T> {
+        // selected_name    ::= `.` suffix [ name_tail ] ;
+        // attribute_name   ::= [ signature ] `'` identifier [ `(` expression `)` ] [ name_tail ] ;
+        // function_name    ::= `(` association_list `)` [ name_tail ] ;
+        // indexed_name     ::= `(` expression { `,` expression } `)` [ name_tail ] ;
+        // slice_name       ::= `(` discrete_range `)` [ name_tail ] ;
+
+        if self.next_is(Dot) {
+            self.start_node(SelectedName);
+            self.expect_token(Dot);
+            self.suffix();
+            self.end_node();
+            self.name_tail();
+        } else if self.next_is(LeftPar) {
+            // Try to differentiate between function calls, indexed names and slices as good as possible:
+            // 1. An `association_list` can be uniquely identified by searching for a '=>' inside the parenthesis
+            // 2. When at least a single comma is found, try parse the contents as an expression list
+            // 3. When the `to` or `downto` keyword is found, parse the content as a slice name
+            //
+            // If none of these apply, it can be either a `subtype_indication` or a single `expression`
+
+            if self.lookahead_in_parens([RightArrow]).is_some() {
+                self.start_node(FunctionCallOrIndexedName);
+                self.expect_token(LeftPar);
+                self.association_list();
+                self.expect_token(RightPar);
+                self.end_node();
+            } else if self.lookahead_in_parens([Comma]).is_some() {
+                self.start_node(FunctionCallOrIndexedName);
+                self.expect_token(LeftPar);
+                self.expression_list();
+                self.expect_token(RightPar);
+                self.end_node();
+            } else if self
+                .lookahead_in_parens([Keyword(Kw::To), Keyword(Kw::Downto)])
+                .is_some()
+            {
+                self.start_node(SliceName);
+                self.expect_token(LeftPar);
+                let closing_paren_distance = self.distance_to_closing_paren();
+                self.range(closing_paren_distance.unwrap());
+                self.expect_token(RightPar);
+                self.end_node();
+            } else {
+                // TODO: subtype_indication or expression?
+                self.start_node(Internal(SubtypeIndicationOrExpressionTokens));
+                self.expect_token(LeftPar);
+                match self.distance_to_closing_paren() {
+                    Some(distance) => self.skip_n(distance),
+                    None => {
+                        self.end_node();
+                        self.eof_err();
+                        return;
+                    }
+                }
+                self.expect_token(RightPar);
+                self.end_node();
+            }
+
+            self.name_tail();
+        } else if is_start_of_attribute_name(self) {
```

Comment:
Due to this little stunt, the `alias_declaration` you pushed yesterday works out of the box :)

Reply:
Clever solution. Nice!